src/rpdk/core/contract/resource_client.py

```diff
@@ -173,15 +173,10 @@ def _update_schema(self, schema):
             for identifier in additional_identifiers
         ]
 
-    def has_writable_identifier(self):
-        for path in self.primary_identifier_paths:
-            if path not in self.read_only_paths:
-                return True
-        for identifier_paths in self._additional_identifiers_paths:
-            for path in identifier_paths:
-                if path not in self.read_only_paths:
-                    return True
-        return False
+    def is_primary_identifier_create_only(self):
```

Comment:
in this case we are checking if any readonly value exists we will return false. The name does not reflect that